runtime/lua/vim/lsp/client.lua

```diff
@@ -603,8 +603,16 @@ local wait_result_reason = { [-1] = 'timeout', [-2] = 'interrupted', [-3] = 'err
 ---
 --- @param ... string List to write to the buffer
 local function err_message(...)
-  api.nvim_err_writeln(table.concat(vim.tbl_flatten({ ... })))
-  api.nvim_command('redraw')
+  local message = table.concat(vim.tbl_flatten({ ... }))
+  if vim.in_fast_event() then
+    vim.schedule(function()
+      api.nvim_err_writeln(message)
+      api.nvim_command('redraw')
+    end)
+  else
+    api.nvim_err_writeln(message)
+    api.nvim_command('redraw')
```

Comment:
instead of checking vim.in_fast_event(), why not always schedule() it?

Reply:
Hmm, good point. 

`err_message` is called while handling errors from user-provided callbacks (`on_init`, ...). Some of them run while the context is not in_fast_event, and can make use of that fact, e.g., call `nvim_err_writeln` without deferring. In that case, the error messages (from user's code and `err_message`) can be in out of order. 

That's unlikely to be an actual problem, but in general I prefer to avoid deferral if possible.

btw there's a similar case here
https://github.com/neovim/neovim/blob/9319d97679053a219b1701d70abce414d0f6a8b0/runtime/lua/vim/lsp/log.lua#L25-L33